src/tools/rbd_mirror/LeaderWatcher.cc

```diff
@@ -90,10 +406,64 @@ void LeaderWatcher::notify_lock_released(Context *on_finish) {
   send_notify(bl, nullptr, on_finish);
 }
 
+void LeaderWatcher::handle_heartbeat(Context *on_notify_ack) {
+  dout(20) << dendl;
+
+  {
+    Mutex::Locker locker(m_lock);
+    if (!m_stopping && !m_releasing && !m_acquiring) {
+      if (m_leader_lock->is_lock_owner()) {
+	dout(0) << "got another leader heartbeat, releasing lock" << dendl;
```

Comment:
I don't think you want to do that here (or below). You can have a transient network hiccup and the Objecter could replay old messages from before this instance acquired the lock. I'd just ignore it since you should have been blacklisted if the lock was broken.